rpc/gnmi/gnmi-specification.md

```diff
@@ -688,20 +690,101 @@ the catalog entry.
+   `<origin, path>` tuple.
+
+If more than one `origin` is to be used within any message, a path in the
+`prefix` MUST NOT be specified, since a prefix applies to all paths within the
+message. In the case that a `prefix` is specified, it MUST specify any required
+`origin`. A single request MUST NOT specify `origin` in both `prefix` and `path`
+fields in any RPC payload messages.
+
+### 2.7.1 Special Values of `origin`
+
+Origin values are agreed out of band to the gNMI protocol. Currently, special
+values are registered within this section of this document. Should additional
+origins be defined, a registry will be defined.
+
+Where the `origin` field is unspecified, its value should default to
+`openconfig`. It is RECOMMENDED that the origin is explictly set.
+
+Where the `origin` field is set to a command line interface format, the path
+should be ignored and the value specified within the `Update` considered as CLI
+input.
+
+CLI configuration is defined as the command line interface text used for
+configuration.  In gNMI, this is encoded as ASCII text.  The format of the ASCII
+text is agreed to out of band from  gNMI.  The string used to identify the CLI
+origin must be formatted as `origin:<nos_cli>`.  Example values might include:
+`xros_cli`, `junos_cli`, `srlinux_cli` or any string which indicates the NOS
+name and the CLI designation.
+
+### 2.7.2 Definition of `origin` for YANG-modelled Data
+
+The `openconfig-extensions:origin` field MAY be utilised to determine the
+origin within which a particular module is instantiated. The specification of
+this extension is within
+[openconfig-extensions.yang](https://github.com/openconfig/public/blob/master/release/models/openconfig-extensions.yang).
+
+It should be noted that `origin` is distinct from `namespace`. Whilst a YANG
+namespace is defined at any depth within the schema tree, an `origin` is
+only used to disambiguate entire schema trees. That is to say, any element
+that is not at the root inherits its `origin` from its root entity, regardless
+of the YANG schema modules that make up that root.
+
+### 2.7.3 Partial Specifications of Origin in Set
+
+If a `Set` RPC specifies `delete`, `update`, or `replace` fields which include
+an `origin` within their `Path` messages, the corresponding change MUST be
+constrained to the specified origin. Particularly:
+
+* `replace` operations MUST only replace the contents of the specified `origin`
+  at the specified path. Origins that are not specified within the `SetRequest`
+  MUST NOT have their contents replaced. In order for a `replace` operation to
+  replace any contents of an `origin` it must be explicitly specified in the
+  `SetRequest`.
+* `delete` operations MUST delete only the contents at the specified path within
+  the specified `origin`. To delete contents from multiple origins, a client
+  MUST specify multiple paths within the `delete` of the `SetRequest`.
```

Comment:
I think we said here that we would note something like:

> These rules apply where origins represent data that doesn't overlap, in some cases (e.g., CLI and OpenConfig) origins may reflect different 'views' on the same data, and thus their interaction is more complex. See <reference to union replace> for more details, and behavioural specifications for such origins.

Reply:
Added